src/Discord/Parts/Embed/Embed.php

```diff
@@ -469,6 +504,26 @@ public function setURL(string $url): self
         return $this;
     }
 
+    /**
+     * Checks to see if an url is valid to be used in embeds.
+     *
+     * @param ?string $url
+     *
+     * @return bool
+     */
+    protected function checkEmbedUrl(?string $url, array $allowed = ['http', 'https', 'attachment']): bool
+    {
+        if (null === $url) return true;
+
+        if ($url === '') return false;
+
+        if (filter_var($url, FILTER_VALIDATE_URL) === false) return false;
+
+        if (! in_array(parse_url($url, PHP_URL_SCHEME), $allowed) return false;
```

Comment:
This is the only required check in this function (tho there is a syntax error) the entire function can be simplified to the following

```suggestion
        return in_array(parse_url($url, PHP_URL_SCHEME), $allowed);
```